samza-core/src/main/java/org/apache/samza/runtime/ContainerLaunchUtil.java

```diff
@@ -141,6 +142,8 @@ static void run(
               samzaEpochId, config);
       MetricsRegistryMap metricsRegistryMap = new MetricsRegistryMap();
 
+      DrainMonitor drainMonitor = new DrainMonitor(coordinatorStreamStore, config);
```

Comment:
I think it might be safer to use a config to control the creation of this drainMonitor. For example, "samza.drainmonitor.enabled" or some thing. We pass in Option.of(null) if this config is set to force. Then we can turn it on for a few jobs first before setting default as true.

Reply:
Added the config and calling the constructor only if the config is true.